src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2018-present Acrolinx GmbH */
+package com.acrolinx.sidebar.pojo.settings;
+
+import java.util.Collections;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class AcrolinxSidebarInitParameterTest
+{
+    public static final String OS_NAME = "os.name";
+    public static final String OS_VERSION = "os.version";
+    public static final String JAVA_VENDOR = "java.vendor";
+    public static final String JAVA_VM_NAME = "java.vm.name";
+    public static final String JAVA_RUNTIME_NAME = "java.runtime.name";
+    public static final String JAVA_RUNTIME_VERSION = "java.runtime.version";
+
+    @Test
+    void toStringTest()
+    {
+        SystemProperties systemProperties = createSystemProperties();
```

Comment:
I am not quite happy with this design. The SystemProperties is mutable, which is ugly. We could add a SystemPropertiesBuilder class and make the SystemProperties immutable, but this adds even more boilerplate code.

Maybe a simpler solution could be to omit this SystemProperties class altogether and instead call `java.lang.System.getProperties()` which returns an `java.util.Properties` instance. We then could make a copy/clone of this Properties to store all the properties before we make changes. Then in the try-block we call `java.lang.System.setProperties(Properties)` to restore the original ones. What do you think?

Reply:
Isn't this what you had implied? Did I get it wrong?